CONTRIBUTING.md

```diff
@@ -0,0 +1,165 @@
+- **refactor** - code change that neither fixes a bug nor adds a feature
+
+**All PRs must include a commit message with the changes description!**
+
+For the initial start, fork the project and use git clone command to download the repository to your computer. A standard procedure for working on an issue would be to:
+
+1. `git pull`, before creating a new branch, pull the changes from upstream. Your master needs to be up to date.
+
+```
+$ git pull
+```
+
+2. Create new branch from `master` like: `doc-548-submit-a-pull-request-section-to-contribution-guide`<br/>
+
+```
+$ git checkout -b [name_of_your_new_branch]
+```
+
+3. Work - commit - repeat ( be sure to be in your branch )
+
+4. Push changes to GitHub
+
+```
+$ git push origin [name_of_your_new_branch]
+```
+
+5. Submit your changes for review
+   If you go to your repository on GitHub, you'll see a `Compare & pull request` button. Click on that button.
+6. Start a Pull Request
+   Now submit the pull request and click on `Create pull request`.
+7. Get a code review approval/reject
+8. After approval, merge your PR
+9. GitHub will automatically delete the branch after the merge is done. (they can still be restored).
+
+## Running
+
+To run Open Runtimes Executor, make sure to install PHP dependencies:
+
+```bash
+docker run --rm --interactive --tty --volume $PWD:/app composer composer install --profile --ignore-platform-reqs
+```
+
+Next start the Docker Compose stack that includes executor server with nessessary networks and volumes:
+
+```bash
+docker compose up -d
+```
+
+You can now use `http://localhost:9800/v1/` endpoint to communicate with Open Runtimes Executor. You can see 'Getting Started' section of README to learn about endpoints.
+
+## Testing
+
+We use PHP framework PHPUnit to test Open Runtimes. Every PR is automatically tested by Travis CI, and tests run for all runtimes. Since this is PHP source code, we also run [Pint](https://github.com/laravel/pint) linter and [PHPStan](https://phpstan.org/) code analysis.
+
+Before running the tests, make sure to install all required PHP libraries:
+
+```bash
+composer install --profile --ignore-platform-reqs
+```
+
+> We run tests in separate Swoole container to ensure unit tests have all nessessary extensions ready.
+
+Once ready, you can test executor.
+
+To run tests, you need to start Docker Compose stack, and then run PHPUnit:
+
+```bash
+docker compose up -d
+# Wait for ~5 seconds for executor to start
+docker run --rm -v $PWD:/app --network executor_runtimes -w /app phpswoole/swoole:4.8.12-php8.0-alpine sh -c \ "composer test"
+```
+
+To run linter, you need to run Pint:
+
+```bash
+composer format
+```
+
+To run static code analysis, you need to run PHPStan:
+
+```bash
+composer check
+```
+
+## Building
+
+To build executor as docker image, run following command:
+
+```bash
+docker build -t openruntimes/executor:dev .
+```
+
+To push image to Docker Hub:
+
+```
+docker push openruntimes/executor:dev
+```
```

Comment:
I think we can remove this section. It's not required for contributing to the project.
The docker-compose file builds the image when running `docker compose up -d` so there isn't a situation where a contributor would need to build and push. Just trying to keep things simple

Reply:
Removing build step from contributing guide.